dist/setup/index.js

```diff
@@ -68324,7 +68324,8 @@ function run() {
                 throw new Error(`Unsupported operating system - ${pkgName} is only released for ${Object.keys(nodePlatformToReleasePlatform).join(", ")}`);
             }
             const releasePlatform = nodePlatformToReleasePlatform[runnerPlatform];
-            const releaseArch = nodeArchToReleaseArch[os.arch()];
+            const osArch = os.arch();
+            const releaseArch = nodeArchToReleaseArch[os.arch()] || osArch;
```

Comment:
When the osArch wasn't in the map we got `undefined` and cached `linux-undefined`.